controllers/idler/aap_idler.go

```diff
@@ -0,0 +1,231 @@
+package idler
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	aapKind       = "AnsibleAutomationPlatform"
+	aapAPIVersion = "aap.ansible.com/v1alpha1"
+)
+
+type aapAPI struct {
+	GVR           schema.GroupVersionResource // AAP GVR
+	ResourceLists []*metav1.APIResourceList   // All available API in the cluster
+}
+
+// ensureAnsiblePlatformIdling checks if there is any long-running pod belonging to an AAP resource and if yes, then it idles the AAP
+// and sends a notification to the user.
+func (r *Reconciler) ensureAnsiblePlatformIdling(ctx context.Context, idler *toolchainv1alpha1.Idler) error {
+
+	// Get all API resources from the cluster using the discovery client. We need it for constructing GVRs for unstructured objects.
+	// Do it here once, so we do not have to list it multiple times before listing/getting every unstructured resource.
+	resourceLists, err := r.DiscoveryClient.ServerPreferredResources()
+	if err != nil {
+		return err
+	}
+
+	// Check if the AAP API is even available/installed
+	aapGVR, err := findGVRForKind(aapKind, aapAPIVersion, resourceLists)
+	if err != nil {
+		return err
+	}
```

Comment:
this could be done only once  - either as part of the controller initialization or using `sync.Once` logic

Reply:
I was thinking about that but then if the API is changed (installed/uninstalled operators) then we will have to restart the member operator manually. The requests seems to be quick enough but we could move it to the initialization (or once) and keep this limitation with installing new operators while the member operator is running in mind.

Reply:
The most realistic risk here if the AAP operator is updated and it introduces some API changes which could break the idler (like introducing another AAP CRD which could be in the middle of the owner chain).